src/common.js

```diff
@@ -17,7 +17,10 @@ export const checkStatus = (response) => {
 
   let error = new Error(response.statusText);
   error.response = response;
-  throw error;
+
+  return response.json()
+    .then(json => Promise.reject(Object.assign(error, json)))
```

Comment:
good idea. I'm thinking of changing the order:
`response.json().catch(() => Promise.reject(error)).then(json => Promise.reject(Object.assign(error, json)));` which will still work but is a little confusing.